src/pybind/mgr/dashboard_v2/restresource.py

```diff
@@ -89,12 +102,12 @@ def default(self, *vpath, **params):
         if not method:
             self._not_implemented(is_element)
 
-        if cherrypy.request.method != 'DELETE':
-            method = _returns_json(method)
-
         if cherrypy.request.method not in ['GET', 'DELETE']:
             method = _takes_json(method)
 
+        if cherrypy.request.method != 'DELETE':
+            method = _returns_json(method)
+
```

Comment:
The order of these decorators shouldn't matter.

Reply:
In this case it matters because otherwise `inspect.getargspec(func).args` will give me the args of the inner function.

Reply:
Exactly. That's why I think you should extract that code into a dedicated decorator. See above.